catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventToXmlElementConverter.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package org.codice.ddf.transformer.xml.streaming.lib;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.xml.sax.Attributes;
+
+/**
+ * A library class used to turn SAX events back into their corresponding XML snippets
+ */
+public class SaxEventToXmlElementConverter {
+
+    private StringBuilder xmlSnippet = new StringBuilder();
```

Comment:
@pklinef @ahoffer I've thought about this some more, and I think if i was going to use an XMLStreamWriter to replace this class, I would have to wrap it in a lot of the same logic. Especially where namespacing and the Attributes are concerned. I think in fact this might be the easiest solution, but I welcome counterpoints.